solr/modules/opentelemetry/src/java/org/apache/solr/opentelemetry/OtelTracerConfigurator.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.opentelemetry;
+
+import io.opentelemetry.opentracingshim.OpenTracingShim;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentracing.Tracer;
+import java.lang.invoke.MethodHandles;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.apache.solr.core.TracerConfigurator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * OpenTracing TracerConfigurator implementation which exports spans to OpenTelemetry in OTLP
+ * format. This impl re-uses the existing OpenTracing instrumentation through a shim, and takes care
+ * of properly closing the backing Tracer when Solr shuts down.
+ */
+public class OtelTracerConfigurator extends TracerConfigurator {
+  private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  // Copy of environment. Can be overridden by tests
+  Map<String, String> currentEnv = System.getenv();
+
+  @Override
+  public Tracer getTracer() {
+    setDefaultIfNotConfigured("OTEL_SERVICE_NAME", "solr");
+    setDefaultIfNotConfigured("OTEL_TRACES_EXPORTER", "otlp");
+    setDefaultIfNotConfigured("OTEL_EXPORTER_OTLP_PROTOCOL", "grpc");
+    setDefaultIfNotConfigured("OTEL_TRACES_SAMPLER", "parentbased_always_on");
+
+    final String currentConfig =
+        String.join(
+            "; ",
+            getCurrentOtelConfig().entrySet().stream()
+                .map(e -> e.getKey() + "=" + e.getValue())
+                .collect(Collectors.toSet()));
```

Comment:
If the objective is to create a String (which I see it is), then it's better to use a `Collectors.joining("; ")` instead of producing a Set to then join it.

Reply:
See new PR #1275, i split it out as a method and sorted by key